### PR TITLE
integrate openshift route plugin with rollouts manager

### PR DESCRIPTION
**Pull request by ishitasequeira**
Integrate [Openshift Route Plugin](https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-openshift) with RolloutsManager.

### CODE REVIEW

File: controllers/configmap.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			Namespace: cr.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name": DefaultRolloutsConfigMapName,
+			},
+		},
+	}
+	trafficRouterPlugins := []types.PluginItem{
+		{
+			Name:     OpenshiftRolloutPluginName,
+			Location: "",
```
Comment (ishitasequeira) [inline controllers/configmap.go:30]: Need to add the https url here.

File: controllers/configmap.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			Location: "file://" + OpenshiftRolloutPluginPath,
+		},
+	}
+	pluginString, err := yaml.Marshal(trafficRouterPlugins)
+	if err != nil {
+		return fmt.Errorf("error marshalling trafficRouterPlugin to string %s", err)
+	}
+	desiredConfigMap.Data["trafficRouterPlugins"] = string(pluginString)
+
+	actualConfiMap := &corev1.ConfigMap{}
```
Comment (anandf) [inline controllers/configmap.go:39]: typo

File: controllers/default.go [hunk 3]
```diff
@@ -17,4 +17,11 @@ const (
 	DefaultRolloutsNotificationSecretName = "argo-rollouts-notification-secret"
 	// DefaultRolloutsServiceSelectorKey is key used by selector
 	DefaultRolloutsSelectorKey = "app.kubernetes.io/name"
+
+	// OpenshiftRolloutPluginName is the plugin name for Openshift Route Plugin
+	OpenshiftRolloutPluginName = "openshift-route-plugin"
```
Comment (anandf) [inline controllers/default.go:22]: Should it be `OpenShift` ?

**Review by anandf — CHANGES_REQUESTED** [review]
posted some minor comments. Otherwise looks ok to merge

File: api/v1alpha1/argorollouts_types.go [hunk 4]
```diff
@@ -43,6 +43,9 @@ type RolloutManagerSpec struct {
 
 	// Version defines Argo Rollouts controller tag (optional)
 	Version string `json:"version,omitempty"`
+
+	// DisableRoutePlugin defines if the Openshift Route Plugin should not be installed on Rollouts Controller
+	DisableRoutePlugin bool `json:"disableRoutePlugin,omitempty"`
```
[the remark this answers is not in the document]
Comment (ishitasequeira) [reply api/v1alpha1/argorollouts_types.go:48]: I like the idea of having a way to extend plugins. I am thinking of below format in that case, let me know what you think.

```
plugins:
    - name: openshift-route-plugin
      enabled: true
      localtion: https://example.com/path/to/rollout-plugin-traffic-controller # if no path is provided, default would be picked from `/plugin` directory.
```

Also, I would like to know how do we prioritize the location value details as users can also change this value in ConfigMap. In the current implementation, we are creating a ConfigMap (if not present already) and updating the ConfigMap with the plugin location only if the ConfigMap does not already contain the plugin name and location. If the plugin information is already present, we do not make any changes (i.e. do not update the location). By adding the location to the rollout-manager spec, how do we prioritize which location value gets picked up?

**Review by anandf — APPROVED** [review]

**Review by iam-veeramalla — APPROVED** [review]
/lgtm 
Thanks @ishitasequeira

**Pull request merged**